### PR TITLE
feat(rpc/v06): add write API methods

### PR DESCRIPTION
**Pull request by kkovaacs**
This PR adds v3 variants to the types used by the gateway client to interface with the `add_transaction` endpoint and maps v3 broadcasted transactions to their gateway-side representation.

The gateway client API is refactored to use the transaction types instead of a long list of parameters and no part of the transactions is now represented as decimal numbers (the gateway seems to accept hexadecimal representations just fine).

Because this PR updates the base (v04) implementation of these methods this also means that now it's possible to add v3 transactions via old JSON-RPC endpoints but I think this is not a problem.

Closes: #1542

### CODE REVIEW

File: crates/gateway-test-fixtures/fixtures/add-transaction/invoke-contract-with-signature.json
Comment (Mirko-von-Leipzig) [inline crates/gateway-test-fixtures/fixtures/add-transaction/invoke-contract-with-signature.json:1]: Has this just always been wrong or did they change representation on the gateway?

File: crates/rpc/src/v04/method/add_deploy_account_transaction.rs [hunk 1]
```diff
@@ -98,30 +100,92 @@ pub async fn add_deploy_account_transaction(
...
-
-    Ok(AddDeployAccountTransactionOutput {
-        transaction_hash: response.transaction_hash,
-        contract_address: response.address,
-    })
+
+    match tx {
+        BroadcastedDeployAccountTransaction::V0V1(
+            tx @ BroadcastedDeployAccountTransactionV0V1 { version, .. },
+        ) if version.without_query_version() == 0 => {
```
Comment (Mirko-von-Leipzig) [inline crates/rpc/src/v04/method/add_deploy_account_transaction.rs:110]: I wonder if there isn't a better abstraction for base version + query flag e.g. an enum + bool which would make this safer to use. Problem for another day though, this is fine for now.

**Review by Mirko-von-Leipzig — APPROVED** [review]

File: crates/rpc/src/v04/method/add_deploy_account_transaction.rs (hunk 1 above)
Comment (kkovaacs) [reply crates/rpc/src/v04/method/add_deploy_account_transaction.rs:110]: I'm sure there is a better representation -- fully agreed.

File: crates/gateway-test-fixtures/fixtures/add-transaction/invoke-contract-with-signature.json
Comment (kkovaacs) [reply crates/gateway-test-fixtures/fixtures/add-transaction/invoke-contract-with-signature.json:1]: The gateway still accepts decimal encoding of these values -- but there's no need to send that, it also accepts hex now (AFAICT this has not been always the case).

Comment (kkovaacs) [reply crates/gateway-test-fixtures/fixtures/add-transaction/invoke-contract-with-signature.json:1]: I just wanted to get rid of the `Deserialize` implementation for these transaction types altogether. We don't actually need those and it seemed to be too much overhead to keep them just for a few tests.

**Pull request merged**